data/content_scripts/privly.js

```diff
@@ -0,0 +1,985 @@
+    if (elements.length > 0) {
+      var extensionMode = elements[0].getAttribute("data-mode");
+      if ( extensionMode === "inactive" ) {
+        return;
+      }
+    }
+
+    privly.createLinks();
+    privly.correctIndirection.run();
+    privly.injectLinks();
+  },
+
+  /**
+   * Listener Function Called when the page is modified with dynamic content
+   * @see privly.addListeners
+   */
+  listenerDOMNodeInserted: function() {
+    "use strict";
+
+    //we check the page a maximum of two times a second
+    if (privly.runPending) {
+      return;
+    }
+
+    privly.runPending = true;
+
+    setTimeout(
+      function(){
+        privly.runPending = false;
+        privly.run();
+      },
+      500);
+  },
+
+  /**
+   * Runs privly once then registers the update listener
+   * for dynamic pages.
+   *
+   * The host page can prevent the non-resize functionality on the page
+   * by defining data-privly-exclude="true" as an attribute on either
+   * the body element.
+   *
+   * @see privly.listenerDOMNodeInserted
+   * @see privly.removeListeners
+   */
+  addListeners: function(){
+
+    "use strict";
+
+    //The content's iframe will post a message to the hosting document.
+    //This listener sets the height  of the iframe according to the messaged
+    //height
+    window.addEventListener("message", privly.resizeIframePostedMessage,
+      false, true);
+
+    privly.runPending = true;
+    setTimeout(
+      function(){
+        privly.runPending = false;
+        privly.run();
+      },
+      100);
+
+
+    // Watch the whole body for changes
+    var target = document.querySelector("body");
+    privly.observer = new MutationObserver(privly.listenerDOMNodeInserted);
+    var config = { childList: true, characterData: true,
+      subtree: true };
+    privly.observer.observe(target, config);
+  },
+
+  /**
+   * Removes event listeners defined by the script. This is primarily
+   * used to deactivate the content script from the Browser's user
+   * interface.
+   * @see privly.addListeners
+   */
+  removeListeners: function(){
+
+    "use strict";
+
+    window.removeEventListener("message", privly.resizeIframePostedMessage,
+      false);
+
+    privly.observer.disconnect();
+    privly.runPending = false;
+  },
+
+  /**
+   * Toggles the display of links and the iframes injected based on the links.
+   */
+  toggleInjection: function() {
+
+    "use strict";
+
+    this.toggleInjectionHelper(document.getElementsByTagName("iframe"), "");
+    this.toggleInjectionHelper(document.getElementsByTagName("a"), "inherit");
+  },
+
+  /*
+   * Helper function for toggling the display of links and iframes
+   *
+   * @param {array} elements Array of HTML elements to be toggled
+   * @param {string} displayStyle CSS style to be used while displaying the HTML elements
+   *
+   */
+  toggleInjectionHelper: function(elements, displayStyle) {
+
+    "use strict"; 
+    var i;
```

Comment:
I know this was directly copied from `privly-firefox`, but having the declaration on it's own line seems silly.